packages/react-core/src/hooks/__tests__/useIsSignedIn.test.ts

```diff
@@ -0,0 +1,90 @@
+import useIsSignedIn from '../useIsSignedIn';
+import { act, renderHook } from '@testing-library/react-hooks';
+import { getCurrentUser } from 'aws-amplify/auth';
+import { GetCurrentUserOutput } from 'aws-amplify/auth';
+import { Hub } from 'aws-amplify/utils';
+
+jest.mock('aws-amplify/auth', () => ({
+  getCurrentUser: jest.fn(),
+}));
```

Comment:
Prefer `jest.spyOn` over `jest.mock`, e.g.
```ts
import * as AuthModule from 'aws-amplify/auth';

....

const getCurrentUserSpy = jest.spyOn(AuthModule, getCurrentUser)
```

Spies retain type definitions of the underlying function which is lost with `jest.mock`